pkg/controlplane/apiserver/options/options.go

```diff
@@ -262,8 +266,41 @@ func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (Comple
 			if err != nil {
 				return CompletedOptions{}, fmt.Errorf("failed to build token generator: %w", err)
 			}
+		case completed.ServiceAccountSigningEndpoint != "":
+			plugin, cache, err := plugin.New(ctx, completed.Authentication.ServiceAccounts.Issuers[0], completed.ServiceAccountSigningEndpoint, 60, true)
+			if err != nil {
+				return CompletedOptions{}, fmt.Errorf("while setting up external-jwt-signer: %w", err)
+			}
+			timedContext, cancel := context.WithTimeout(ctx, time.Duration(5)*time.Second)
```

Comment:
cast isn't necessary; an error here will fail apiserver startup, so I'd suggest at least 10 seconds (default dial timeout for regular http clients)

```suggestion
			timedContext, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Reply:
Done